C code gen: undeclared alias type for a dynamic array when using a reference to it

### Describe the bug

When referencing a dynamic array instantiated via a type alias, V compiler generates invalid C code.

### Reproduction Steps

```v
type Array = []int

fn main() {
	a := Array{}
	n := &a
}
```

### Expected Behavior

success

### Current Behavior

```c
warning: tcc compilation failed, falling back to cc (this is much slower)
================== C compilation error (from tcc): ==============
cc: /tmp/v_501/cgen_undeclared.01KWMK7Q88Y1RNVMBEAGJEA459.tmp.c:4112: error: 'main__Array' undeclared
================== C compiler bug report ==============
Generated C lines sent from /tmp/v_501/cgen_undeclared.01KWMK7Q88Y1RNVMBEAGJEA459.tmp.c:6910:
    6905 | 
    6906 | #line 3 "/cgen_undeclared.v"
    6907 | VV_LOC void main__main(void) {
    6908 | 
    6909 | #line 4 "/cgen_undeclared.v"
>   6910 |      main__Array *a = HEAP(main__Array, (builtin____new_array_with_default_noscan(0, 0, sizeof(int), 0)));
    6911 | 
    6912 | #line 5 "/cgen_undeclared.v"
    6913 |      int* n = &(*(int*)builtin__array_get((*(a)), 0));
    6914 | }
    6915 | 
Corresponding V lines sent from /cgen_undeclared.v:4:
       1 | type Array = []int
       2 | 
       3 | fn main() {
>      4 |      a := Array{}
       5 |      n := &a[0]
       6 | }
=======================================================
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.5.1 9dce771

### Environment details (OS name and version, etc.)

not relevant
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/markused/walker.v
+++ b/vlib/v/markused/walker.v
@@ -726,6 +726,9 @@ pub fn (mut w Walker) stmt(node_ ast.Stmt) {
 			}
 		}
 		ast.AssignStmt {
+			for t in node.left_types {
+				w.mark_by_type(t)
+			}
 			w.exprs(node.left)
 			w.exprs(node.right)
 		}

--- a/vlib/v/tests/aliases/alias_array_reference_test.v
+++ b/vlib/v/tests/aliases/alias_array_reference_test.v
@@ -1,0 +1,8 @@
+type Arr = []int
+
+fn test_alias_array_reference() {
+	mut a := Arr{len: 1}
+	a[0] = 69
+	p := &a[0]
+	assert *p == 69
+}

